sos/utilities.py

```diff
@@ -226,22 +226,22 @@ def _check_poller(proc):
         else:
             reader = FakeReader(p, binary)
 
-        if poller:
-            while reader.running:
-                _check_poller(p)
-        else:
-            try:
+        try:
+            if poller:
+                while reader.running:
+                    _check_poller(p)
+            else:
                 # override timeout=0 to timeout=None, as Popen will treat the
                 # former as a literal 0-second timeout
                 p.wait(timeout if timeout else None)
-            except Exception:
-                p.terminate()
-                if to_file:
-                    _output.close()
-                # until we separate timeouts from the `timeout` command
-                # handle per-cmd timeouts via Plugin status checks
-                return {'status': 124, 'output': reader.get_contents(),
-                        'truncated': reader.is_full}
+        except Exception:
+            p.terminate()
+            if to_file:
+                _output.close()
+            # until we separate timeouts from the `timeout` command
+            # handle per-cmd timeouts via Plugin status checks
+            return {'status': p.poll(), 'output': reader.get_contents(),
```

Comment:
For this we need to return 124 for any timeouts, as many other parts of sos rely on that to identify the timeout as that is the standard exit code.

If `Popen()` is changing this internally, we need to change it back to what's expected.

Reply:
Sure, I agree with you, I will change it in update.